### PR TITLE
Add ability to specify renew lease ID in POST body.

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: http/handler.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func Handler(core *vault.Core) http.Handler {
 	mux.Handle("/v1/sys/seal", handleSysSeal(core))
 	mux.Handle("/v1/sys/step-down", handleSysStepDown(core))
 	mux.Handle("/v1/sys/unseal", handleSysUnseal(core))
-	mux.Handle("/v1/sys/renew/", handleLogical(core, false, nil))
+	mux.Handle("/v1/sys/renew", handleLogical(core, false, nil))
```
Comment (vishalnayak) [inline http/handler.go]: Will this not route the `/v1/sys/renew/<urlparam>` to `handleLogical` with `dataOnly` set to `true`?

File: vault/logical_system.go [hunk 2]
```diff
@@ -263,9 +263,13 @@ func NewSystemBackend(core *Core, config *logical.BackendConfig) logical.Backend
 			},
 
 			&framework.Path{
-				Pattern: "renew/(?P<lease_id>.+)",
+				Pattern: "renew" + framework.OptionalParamRegex("urllease_id"),
 
 				Fields: map[string]*framework.FieldSchema{
+					"urllease_id": &framework.FieldSchema{
```
Comment (vishalnayak) [inline vault/logical_system.go]: [Optional] I see the rationale behind `urllease_id` (urltoken, urlaccessor, etc), but somehow `url_lease_id` also seems like a better fit here.

File: vault/policy_store.go [hunk 3]
```diff
@@ -301,6 +301,10 @@ path "sys/capabilities-self" {
     capabilities = ["update"]
 }
 
+path "sys/renew" {
```
Comment (vishalnayak) [inline vault/policy_store.go]: Can we pull the default policy into a constant above?

File: vault/logical_system.go (hunk 2 above)
Comment (jefferai) [inline vault/logical_system.go]: It's internal either way  :-)

File: vault/policy_store.go (hunk 3 above)
Comment (jefferai) [inline vault/policy_store.go]: Done.

File: vault/logical_system.go (hunk 2 above)
Comment (jefferai) [inline vault/logical_system.go]: Done.

**Comment by vishalnayak** [comment]
LGTM!

File: vault/logical_system_test.go [hunk 4]
```diff
@@ -331,6 +330,64 @@ func TestSystemBackend_renew(t *testing.T) {
...
+		t.Fatalf("err: %v", err)
+	}
+	if resp2.IsError() {
+		t.Fatalf("got an error")
+	}
+	if resp2.Data == nil {
+		t.Fatal("nil data")
+	}
+	if resp.Secret.TTL != 180*time.Second {
+		t.Fatal("bad lease duration: %v", resp.Secret.TTL)
```
Comment (vishalnayak) [inline vault/logical_system_test.go]: s/Fatal/Fatalf

File: vault/logical_system_test.go [hunk 5]
```diff
@@ -331,6 +330,64 @@ func TestSystemBackend_renew(t *testing.T) {
...
```
Comment (vishalnayak) [inline vault/logical_system_test.go]: s/Fatal/Fatalf

File: vault/logical_system_test.go (hunk 4 above)
Comment (jefferai) [inline vault/logical_system_test.go]: Thanks, fixed.

**Pull request merged**